### ISSUE TITLE
phpdoc for integration methods

### ISSUE THREAD

**Issue by staabm** [itsgoingd/clockwork#394]
would be great to have phpdocs for the most used methods which are required for integration with the host applications, e.g.

- `resolveAsCommand`
- `resolveAsQueueJob`

at best we could have psalm or phpstan like phpdocs, which are pretty precise about the contents of the arrays involved

**Comment by itsgoingd** [comment]
yeah, i can see how writing the integration might be challenging since the clockwork internals are not very well documented

not sure i want to commit to phpdoc tho, i like to keep the comments simple, might add api docs to the website instead

**Comment by staabm** [comment]
> ot sure i want to commit to phpdoc tho, i like to keep the comments simple, might add api docs to the website instead

having only the types defined in the php sources (without further full-text description) would already be enough, as our static analysis jobs (and IDE) would do the ret
